File: Programming-Basics-With-JavaScript/Homeworks-And-Labs/05.While-Loop/01-lab/08.graduation.js
function graduation(input) {

    const name = input[0];
    let index = 1;
    let annualEvaluation = Number(input[index]);

    let gradeCount = 1;
    let sumEvaluation = 0;
    let excludedCount = 0;
    let isExcluded = false;

    while (gradeCount <= 12) {
        if (annualEvaluation < 4) {
            excludedCount += 1;
            if (excludedCount > 1) {
                isExcluded = true;
                break;
            }
            continue;
        }

        sumEvaluation += annualEvaluation;
        gradeCount += 1;
        index++;
        annualEvaluation = Number(input[index]);
    }

    const averageEvaluation = sumEvaluation / 12;

    if (isExcluded) {
        console.log(`${name} has been excluded at ${gradeCount} grade`);
    } else {
        console.log(`${name} graduated. Average grade: ${averageEvaluation.toFixed(2)}`);
    }
}
graduation(["Mimi", "5", "6", "5", "6", "5", "6", "6", "2", "3"]);

File: Python_Basics/Homeworks-And-Labs/05.while_loop/01.lab/08.graduation.py
name = input()

grade_count = 1
sum_evaluation = 0
excluded_count = 0
is_excluded = False

while grade_count <= 12:
    annual_evaluation = float(input())
    if annual_evaluation < 4:
        excluded_count += 1
        if excluded_count > 1:
            is_excluded = True
            break
        continue

    sum_evaluation += annual_evaluation
    grade_count += 1
   

average_evaluation = sum_evaluation / 12

if is_excluded:
    print(f'{name} has been excluded at {grade_count} grade')
else:
    print(f'{name} graduated. Average grade: {average_evaluation:.2f}')
